lib/chef/mixin/windows_architecture_helper.rb

```diff
@@ -43,6 +43,14 @@ def wow64_architecture_override_required?(node, desired_architecture)
       end
 
       def with_os_architecture(node)
+        node ||= begin
```

Comment:
mrrrrr. I guess sometime we should method to not eat a whole node and make this code you're adding a util method.